File: CS/Models/DocumentPreviewViewModel.cs
using DevExpress.Mvvm;
using DevExpress.XtraReports.UI;

namespace PassDataFromViewModelToReport.Models {
    public class DocumentPreviewViewModel : BindableBase {
        public XtraReport Report { get; }
        public DocumentPreviewViewModel(XtraReport report) {
            Report = report;
        }
    }
}

File: VB/Models/DocumentPreviewViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.XtraReports.UI

Namespace PassDataFromViewModelToReport.Models
	Public Class DocumentPreviewViewModel
		Inherits BindableBase

		Public ReadOnly Property Report() As XtraReport
		Public Sub New(ByVal report As XtraReport)
			Me.Report = report
		End Sub
	End Class
End Namespace
